app/models/logs_search.rb

```diff
@@ -22,16 +21,24 @@ def validate_term
     end
   end
 
+  def validate_presence
+    if term.length == 0
+      self.errors.add(:search_term, 'cannot be empty')
+    end
+  end
+
   def validate_username
-    unless term.length == 5 || term.length == 6
-      self.errors.add(:term, 'must be 5 or 6 characters')
+    unless term.length == 5 || term.length == 6 || term.length == 0
```

Comment:
I don't understand why this test is here and in `validate_presence`.

Reply:
This is to skip adding the error `'must be 5 or 6 characters'` if the search term is empty. This avoids the duplicate error message.

Achieves the same as line 37:
```
  def validate_ip
    validate_presence if term.empty? && return
```